File: src/jvm/desig/pattern/visitor/IncomeBill.java
package jvm.desig.pattern.visitor;

/**
 * 收入账单
 */
public class IncomeBill implements Bill {
    private String item;
    private double amount;

    String getItem() {
        return item;
    }

    double getAmount() {
        return amount;
    }

    IncomeBill(String item, double amount) {

        this.item = item;
        this.amount = amount;
    }

    @Override
    public void accept(AccountBookView accountBookView) {
        accountBookView.view(this);
    }
}

File: src/kt/desig/pattern/visitor/IncomeBill.kt
package kt.desig.pattern.visitor

/**
 * 支出单子
 */
class IncomeBill : Bill {
    private var item: String = ""
    internal var amount: Double = 0.0

    internal constructor(item: String, amount: Double) {
        this.item = item
        this.amount = amount
    }

    fun getItem(): String {
        return item
    }

    fun getAmount(): Double {
        return amount
    }

    override fun accept(accountBookView: AccountBookView) {
        accountBookView.view(this)
    }
}
